get.js

```diff
@@ -17,51 +52,70 @@ module.exports = function getRecommendations (config, options) {
     var defaultCurrency = settings.defaultCurrency || config.defaultCurrency
     var defaultLanguage = settings.defaultLanguage || config.defaultLanguage
 
-    if (!_.isArray(seed)) {
-      seed = [seed]
-    }
-
-    // convert { category: 'x' } -> { type: 'c', id: 'x' }
-    // hopefully, the API will add support for the former
-    // at which point we can remove this mapping
-    seed = _.map(seed, function (s) {
-      if (_.isObject(s) && s.category) {
-        return { type: 'c', id: s.category }
-      } else {
-        return s
+    if (seed !== 'all') {
+      if (!_.isArray(seed)) {
+        seed = [seed]
       }
-    })
-
-    var data = { h: seed }
 
-    if (rules) {
-      _.assign(data, { rules: rules })
+      // convert { category: 'x' } -> { type: 'c', id: 'x' }
+      // hopefully, the API will add support for the former
+      // at which point we can remove this mapping
+      seed = _.map(seed, function (s) {
+        if (_.isObject(s) && s.category) {
+          return { type: 'CATEGORY', value: s.category }
+        } else {
+          return { type: 'PRODUCT', value: s }
+        }
+      })
+    } else {
+      seed = null
     }
 
     return getLocale({
       uv: options.uv,
       defaultCurrency: defaultCurrency,
       defaultLanguage: defaultLanguage
     }).then(function (locale) {
-      var requestUrl = [
-        url,
-        trackingId,
-        '?strategy=' + strategy,
-        '&id=' + visitorId,
-        '&n=' + limit,
-        '&experienceId=' + config.experienceId,
-        '&iterationId=' + config.iterationId,
-        '&variationId=' + config.variationId,
-        '&locale=' + locale
-      ].join('')
+      var variables = {
+        trackingId: trackingId,
+        contextId: visitorId,
+        experienceId: config.experienceId,
+        items: limit,
+        strategy: [{ name: strategy }],
+        seed: seed,
+        rules: rules,
+        locale: locale
+      }
 
-      var request = http.post(requestUrl, JSON.stringify(data), { timeout: timeout })
-      return request.then(function (result) {
+      return qubitApi.query(query, variables, {
+        timeout: timeout,
+        url: url
+      }).then(function (result) {
         if (result) {
-          var recs = JSON.parse(result)
-          var items = _.get(recs, 'result.items')
-
+          var items = _.get(
+            result,
+            'data.property.visitor.productRecommendations'
+          )
           if (items && items.length) {
+            // Convert back to native Recs API format to support legacy experiences
+            items = _.map(items, function (item) {
+              item.id = item.product.product_id
```

Comment:
Can we use an empty object to return from this function, rather than mutating these items.